### PR TITLE
FeatureDataSource: output cache hit rate to logger.debug() upon close

### PR DESCRIPTION
**Pull request by droazen**
Modified FeatureCache to track the number of cache hits/misses over the
lifetime of each FeatureDataSource. Upon close, output these statistics
to logger.debug().

Tracking this information introduces almost zero overhead, but should be
useful on an ongoing basis as we add new tools to GATK4 and profile them.

### CODE REVIEW

**Comment by droazen** [comment]
@akiezun Quickie review for you

File: src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java [hunk 1]
```diff
@@ -559,6 +593,18 @@ private void closeOpenIterationIfNecessary() {
     }
 
     /**
+     * Print statistics about the cache hit rate for debugging
+     */
+    private void printCacheStatistics() {
+        final int totalQueries = queryCache.getNumCacheHits() + queryCache.getNumCacheMisses();
+        logger.debug(String.format("FeatureDataSource for %s: Cache hit rate was %.2f%% (%d out of %d total queries)",
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java]: FeatureDataSource gets printed twice on every line

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java]: provide the logical name for each feature input, if available.

**Comment by akiezun** [comment]
lgtm, 2 small improvements and it's good to merge

File: src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java (hunk 1 above)
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java]: Done and done

**Pull request merged**